### PR TITLE
Initialize properly plot canvas in plot's subclasses

### LINKED ISSUE #6210: bokehjs doesn't work in notebook
A side effect of bad programming practice in gmap's code and PR #5976.

### PR DESCRIPTION
**Pull request by mattpap**
fixes #6210

### CODE REVIEW

**Comment by bryevdv** [comment]
What do we need to get some form of minimal notebook testing back in our Travis builds?

File: bokehjs/src/coffee/models/plots/plot.coffee [hunk 1]
```diff
@@ -103,7 +103,7 @@ export class Plot extends LayoutDOM
       title = if isString(@title) then new Title({text: @title}) else @title
       @add_layout(title, @title_location)
 
-    @_plot_canvas = new PlotCanvas({plot: @})
+    @_init_plot_canvas()
```
Comment (bryevdv) [inline bokehjs/src/coffee/models/plots/plot.coffee]: I think `@_plot_canvas = @_init_plot_canvas()` (i.e. having the method return the canvas)  might be nicer than relying on the side effect of setting the property.

**Comment by mattpap** [comment]
> What do we need to get some form of minimal notebook testing back in our Travis builds?

I don't know. notebooks didn't work for a long time, so there may be several issues to fix.

**Comment by bryevdv** [comment]
> I don't know. notebooks didn't work for a long time, so there may be several issues to fix.

I'd settle for a couple of canary image diffs that hard fail, at this point. 

Do you have any thoughts about the other comment? I'm happy to make the change, if you are busy (but agree)

File: bokehjs/src/coffee/models/plots/plot.coffee (hunk 1 above)
Comment (bryevdv) [inline bokehjs/src/coffee/models/plots/plot.coffee]: For that matter, I'd prefer not to have a methods at all, can't we add a class variable for the type? Then
```
@_plot_canvas = new @canvas_type({plot: @})
```
ping @mattpap

**Comment by mattpap** [comment]
> I'd settle for a couple of canary image diffs that hard fail, at this point.

This is not a question of scope, but just making our testing infrastructure work with notebook again. I don't remember what was the problem, I need to refresh my memory on this. I just suspect that there was an issue with detecting when notebook evaluated cells and bokehjs finished rendering, so possibly existing heuristic or an improved one will help with this.

**Comment by bryevdv** [comment]
> This is not a question of scope, but just making our testing infrastructure work with notebook again. I don't remember what was the problem, I need to refresh my memory on this. I just suspect that there was an issue with detecting when notebook evaluated cells and bokehjs finished rendering, so possibly existing heuristic or an improved one will help with this.

right getting notebooks tested regularly again is definitely for a separate issue. I'd like to get this merged an in a dev build. Can you address the comment about avoiding side effects?

**Comment by mattpap** [comment]
> Can you address the comment about avoiding side effects?

@bryevdv, I already did.

**Comment by bryevdv** [comment]
@bryevdv, I already did.

I missed it, sorry! merging now

**Pull request merged**